pkgs/development/python-modules/torch/default.nix

```diff
@@ -210,13 +207,14 @@ in buildPythonPackage rec {
   # when using the different backends. Gloo's GPU support isn't great, and MPI and CUDA can't be used at the same time
   # without extreme care to ensure they don't lock each other out of shared resources.
   # For more, see https://github.com/open-mpi/ompi/issues/7733#issuecomment-629806195.
-  preConfigure = lib.optionalString cudaSupport ''
+  preConfigure = lib.optionalString cudaSupport (''
     export TORCH_CUDA_ARCH_LIST="${gpuTargetString}"
-    export CUDNN_INCLUDE_DIR=${cudnn.dev}/include
-    export CUDNN_LIB_DIR=${cudnn.lib}/lib
     export CUPTI_INCLUDE_DIR=${cudaPackages.cuda_cupti.dev}/include
     export CUPTI_LIBRARY_DIR=${cudaPackages.cuda_cupti.lib}/lib
-  '' + lib.optionalString rocmSupport ''
+  '' + lib.optionalString (cudaPackages ? cudnn) ''
+    export CUDNN_INCLUDE_DIR=${cudnn.dev}/include
+    export CUDNN_LIB_DIR=${cudnn.lib}/lib
+  '') + lib.optionalString rocmSupport ''
```

Comment:
Oof I'm not sure we've got a single autoformatter that would keep this intact. I know pytorch's formatting is mostly manual, but I would prefer if local changes were moving us towards where autoformatters would keep more stuff in place

Would `optionalString cudaSupport && cudaPackages ? cudnn` be too much?